### PR TITLE
ROX-19344: Add tabs and vuln deferral config container

### PR DESCRIPTION
**Pull request by dvail**
## Description

Adds the tab structure (Vulnerabilities tab only) and basic page layout.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Load the deferral config page:

image
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7816** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7816" target="_blank">Graphite</a> 
    * **PR #7825** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7825" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7825?utm_source=stack-comment).

File: ui/apps/platform/src/Containers/DeferralConfiguration/DeferralConfigurationPage.tsx [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+import { PageSection, Tab, Tabs, Title } from '@patternfly/react-core';
 
 import PageTitle from 'Components/PageTitle';
+import useURLStringUnion from 'hooks/useURLStringUnion';
+import VulnerabilitiesConfigurationContainer from './VulnerabilitiesConfigurationContainer';
+
+const deferralConfigurationCategories = ['Vulnerabilities'] as const;
 
 function DeferralConfigurationPage() {
+    const [category, setCategory] = useURLStringUnion('category', deferralConfigurationCategories);
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/DeferralConfiguration/DeferralConfigurationPage.tsx:11]: Oh, I didn't know about this hook. I was initially a little confused about what it did from just the name

File: ui/apps/platform/src/Containers/DeferralConfiguration/VulnerabilitiesConfigurationContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Button,
+    Divider,
+    PageSection,
+    Split,
+    SplitItem,
+    Text,
+    Title,
+} from '@patternfly/react-core';
+
+function VulnerabilitiesConfigurationContainer() {
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/DeferralConfiguration/VulnerabilitiesConfigurationContainer.tsx:12]: nit: I think we could just say `VulnerabilitiesConfiguration` here. The `Container` part doesn't seem as beneficial

**Review by sachaudh — APPROVED** [review]

**Comment by sachaudh** [comment]
@zhenpesky @mansursyed In the mock for the Deferral Configuration page, we show these disabled tabs. Are those what we expect to add in the future or are they actually just placeholder text (not actually relevant). Wasn't sure what they were doing under the "Deferral" configuration

https://docs.google.com/presentation/d/1PCYmIS9SkRr9lnZdPJ4bdEkFD3cdq2vYZExq0P_ltn4/edit#slide=id.g25da666987a_0_10

**Comment by mansursyed** [comment]
> @zhenpesky @mansursyed In the mock for the Deferral Configuration page, we show these disabled tabs. Are those what we expect to add in the (near/far) future or are they actually just placeholder text (not actually relevant)? Wasn't sure what they were doing under the "Deferral" configuration. Just wanted to know if having a tab is reasonable now or if it makes more sense to do it later on
> 
> https://docs.google.com/presentation/d/1PCYmIS9SkRr9lnZdPJ4bdEkFD3cdq2vYZExq0P_ltn4/edit#slide=id.g25da666987a_0_10

I think this is something that can be done later on. The deferral mechanism may change more in the future once we've fully fleshed out the other use cases so we don't need to invest time in building the tabs into the page now. @sachaudh

File: ui/apps/platform/src/Containers/DeferralConfiguration/DeferralConfigurationPage.tsx (hunk 1 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/DeferralConfiguration/DeferralConfigurationPage.tsx:11]: Hmm we could rename it, what would be a better name?
`useURLStringChoice`
`useURLEnum`
`useURLOption`

**Pull request merged**